viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -333,97 +371,81 @@ void SLAMServiceImpl::GetLatestSampledPointCloudMapString(
+        for (int pixel_x = 0; pixel_x < width; pixel_x++) {
+            // Get byte index associated with pixel
+            int pixel_index = pixel_x + pixel_y * width;
+            int byte_index = pixel_index * bytesPerPixel;
+
+            // We assume we are running on a little-endian system, so the ARGB
+            // order is reversed
+            pixelColorARGB color;
+            color.A = image_data[byte_index + 3];
+            color.R = image_data[byte_index + 2];
+            color.G = image_data[byte_index + 1];
+            color.B = image_data[byte_index + 0];
+
+            // Skip pixel if it contains empty data (default color)
+            if (checkIfEmptyPixel(color)) {
+                continue;
+            }
 
-    // Sample the image based on the number of pixels. Output is the number of
-    // pixels to skip. skip_count will reduce the size of the PCD to under 32
-    // MB, with additional tuning provided by the samplingFactor. If the PCD
-    //  would already be smaller than 32MB, do not sample the image.
-    // When moving to streaming this behavior may change.
-    int skip_count = (size_data * pixelBytetoPCDByte) / maximumGRPCByteLimit *
-                     samplingFactor;
-    if (skip_count == 0) {
-        skip_count = 1;
-    }
-
-    std::string data_buffer;
-
-    // Loop to sample data and reduce resolution. Increments multiplied
-    // by 4 to represent 4 bytes per pixel
-    for (int i = 0; i < size_data; i += skip_count * 4) {
-        // skip pixels that are not in our map(black/past walls)
-        // this check represents [102,102,102]
-        if ((data_vect[i + 0] == defaultCairosEmptyPaintedSlice) &&
-            (data_vect[i + 1] == defaultCairosEmptyPaintedSlice) &&
-            (data_vect[i + 2] == defaultCairosEmptyPaintedSlice))
-            continue;
-
-        // Determine probability based on color pixel
-        int prob = viam::ViamColorToProbability((int)data_vect[i + 2]);
-        if (prob == 0) continue;
-
-        num_points++;
-
-        int pixel_index = i / 4;
-        int pixel_x = pixel_index % width;
-        int pixel_y = pixel_index / width;
-
-        // Convert pixel location to pointcloud point in meters
-        float x_pos = (pixel_x - painted_slices->origin.x()) * kPixelSize;
-        // Y is inverted to match output from getPosition()
-        float y_pos = -(pixel_y - painted_slices->origin.y()) * kPixelSize;
-        // 2D SLAM so Z is set to 0
-        float z_pos = 0;
-
-        // Turn the map point into a vector to perform transformations with.
-        // Current transformation rotates coordinates to match slam service
-        // expectation (XZ plane)
-        Eigen::Vector3d map_point(x_pos, y_pos, z_pos);
-        auto rotated_map_point = pcdRotation * map_point;
+            // Determine probability based on color pixel and skip if 0
+            int prob = calculateProbabilityFromColorChannels(color);
```

Comment:
```suggestion
            // Determine probability based on the pixel color and skip if it is 0
            int prob = CalculateProbabilityFromColorChannels(pixel_color);
```

Reply:
Corrected comment and uppercase function name